internal/ingress/controller/store/store.go

```diff
@@ -532,42 +532,47 @@ func New(
 			// updates to configuration configmaps can trigger an update
 			if key == configmap || key == tcp || key == udp {
 				recorder.Eventf(cm, corev1.EventTypeNormal, "CREATE", fmt.Sprintf("ConfigMap %v", key))
+
 				if key == configmap {
 					store.setConfig(cm)
 				}
-				updateCh.In() <- Event{
-					Type: ConfigurationEvent,
-					Obj:  obj,
-				}
+			}
+
+			updateCh.In() <- Event{
+				Type: ConfigurationEvent,
+				Obj:  obj,
 			}
 		},
 		UpdateFunc: func(old, cur interface{}) {
-			if !reflect.DeepEqual(old, cur) {
-				cm := cur.(*corev1.ConfigMap)
-				key := k8s.MetaNamespaceKey(cm)
-				// updates to configuration configmaps can trigger an update
-				if key == configmap || key == tcp || key == udp {
-					recorder.Eventf(cm, corev1.EventTypeNormal, "UPDATE", fmt.Sprintf("ConfigMap %v", key))
-					if key == configmap {
-						store.setConfig(cm)
-					}
+			if reflect.DeepEqual(old, cur) {
+				return
+			}
 
-					ings := store.listers.IngressWithAnnotation.List()
-					for _, ingKey := range ings {
-						key := k8s.MetaNamespaceKey(ingKey)
-						ing, err := store.getIngress(key)
-						if err != nil {
-							klog.Errorf("could not find Ingress %v in local store: %v", key, err)
-							continue
-						}
-						store.syncIngress(ing)
-					}
+			cm := cur.(*corev1.ConfigMap)
+			key := k8s.MetaNamespaceKey(cm)
+			// updates to configuration configmaps can trigger an update
+			if key == configmap || key == tcp || key == udp {
+				recorder.Eventf(cm, corev1.EventTypeNormal, "UPDATE", fmt.Sprintf("ConfigMap %v", key))
+			}
 
-					updateCh.In() <- Event{
-						Type: ConfigurationEvent,
-						Obj:  cur,
-					}
+			if key == configmap {
+				store.setConfig(cm)
+			}
+
+			ings := store.listers.IngressWithAnnotation.List()
+			for _, ingKey := range ings {
+				key := k8s.MetaNamespaceKey(ingKey)
+				ing, err := store.getIngress(key)
+				if err != nil {
+					klog.Errorf("could not find Ingress %v in local store: %v", key, err)
+					continue
 				}
+				store.syncIngress(ing)
+			}
```

Comment:
Why do you have to force this here? If ingresses changed, controller will eventually be aware of that, no?

Reply:
> Why do you have to force this here? 

Because right now we don't have an intelligent way to detect if an ingress references a configmap in one or more annotation.